cluster-provision/gocli/cmd/run.go

```diff
@@ -574,19 +575,37 @@ func run(cmd *cobra.Command, args []string) (retErr error) {
 	// the VM console from the container without ssh
 	qemuArgs += " -serial pty"
 
+	arch := runtime.GOARCH
+	var qemuDevice string
+
+	// Use virtio-net-ccw device incase of s390x Architecture.
+	if arch == "s390x" {
+		qemuDevice = "virtio-net-ccw"
+	} else {
+		qemuDevice = "virtio-net-pci"
+	}
+	
 	wg := sync.WaitGroup{}
 	wg.Add(int(nodes))
 	// start one vm after each other
 	macCounter := 0
 	for x := 0; x < int(nodes); x++ {
 
 		nodeQemuArgs := qemuArgs
+		nodeQemuMonitorArgs := ""
 
 		for i := 0; i < int(secondaryNics); i++ {
 			netSuffix := fmt.Sprintf("%d-%d", x, i)
 			macSuffix := fmt.Sprintf("%02x", macCounter)
 			macCounter++
-			nodeQemuArgs = fmt.Sprintf("%s -device virtio-net-pci,netdev=secondarynet%s,mac=52:55:00:d1:56:%s -netdev tap,id=secondarynet%s,ifname=stap%s,script=no,downscript=no", nodeQemuArgs, netSuffix, macSuffix, netSuffix, netSuffix)
+			// Secondary network devices are added after VM is started (hot-plug) using qemu monitor to avoid 
+			// primary network interface to be named other than eth0. This is mainly required for s390x, as 
+		    // otherwise if primary interface is other than eth0, it can't get the IP from dhcp server.
+			if qemuDevice == "virtio-net-ccw" {
```

Comment:
see above, then this would be more obvious.

Reply:
Yes, updated.